### PR TITLE
Deployment improvements

### PR DESCRIPTION
**Pull request by iros**
Please review
Here to keep track of progress and todo items.

Making many deployment improvements such as:

- [x] support running different environments 
- [x] fix AbortionError intermittent issue
- [x] migrate to Docker-based local running
- [x] conform to linting rules from git-hooks
- [x] ensure tests are passing
- [x] support deploying to staging and sandbox
- [x] simplify the credential file sharing. Right now we need to pass it to the ./deploy.sh script, but we can't pass it to ./run.sh if it's outside the repo because ./run.sh is also used from the docker container on the server, thus it does not have access to whatever your key folder is. Locally though, it'd be nice to just pass a KEY_FILE too. Could check for the presence of `cred.json` and then just use it, but it seems inconsistent (plus once we move it once, we never remove it).

README.md is updated as I go.
You can setup and run a local server more easily now using Docker.
Changed from local run environment to Docker due to challenges deploying a flex python environment (which now requires a Dockerfile). Switched to custom runtime.

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
```
Comment (iros) [inline deploy.sh]: This has not been run yet!

**Comment by iros** [comment]
Examples: 
*Deployment:*
```
KEY_FILE=../mlab-keys/mlab-sandbox-pipeline-bigtable\@mlab-sandbox.iam.gserviceaccount.com-creds.json ./deploy.sh sandbox
```

*Local run:*
```
docker build -t mlab-vis-api .
docker run -p 8080:8080 -e API_MODE=production mlab-vis-api
```
Note that your `cred.json` file should match the API_MODE (for project id etc.)

I'd really love to get rid of this cred.json file, so I'm going to investigate that next time.

**Comment by iros** [comment]
Simplified the credential passing. Now mounting a volume for local docker running/testing, and copying KEY_FILE temporarily for deployment.
Also changed local test routine to use a test docker container.

File: deploy.sh (hunk 1 above)
Comment (iros) [reply deploy.sh]: It has now. All is good.

**Comment by critzo** [comment]
@iros Sorry for the delay on this. LGTM!

**Pull request merged**